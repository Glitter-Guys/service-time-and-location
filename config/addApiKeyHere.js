// Rename this file to apiKeys.js
// Get a Google Maps key here: https://developers.google.com/maps/documentation/static-maps/
// And replace it below
// It should then automatically be imported to components/map.jsx

exports.googleMapsKey = 'addyourkeyhere';
