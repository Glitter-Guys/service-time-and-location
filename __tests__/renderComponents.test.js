import React from 'react';
import expect from 'expect';
import { shallow, mount, render } from 'enzyme';
import WhereWhen from './../client/components/whereWhen.jsx';
import Where from './../client/components/where.jsx';
import When from './../client/components/when.jsx';
import Map from './../client/components/map.jsx';
import { googleMapsKey } from './../config/apiKeys';

describe('<WhereWhen />', () => {
 it('should be defined', () => {
   expect(WhereWhen).toBeDefined();
 });
});

describe('<When />', () => {
 it('should be defined', () => {
   expect(When).toBeDefined();
 });
 
});

describe('<Where />', () => {
 it('should be defined', () => {
   expect(Where).toBeDefined();
 });
});

describe('<Map />', () => {
  const dummyWhereData = {
    address1: "30 Nowhere St.",
    city: "Sometown",
    state: "CA"
  }
  const dummyLat = 37.75;
  const dummyLon = -122.419998;
  const mapWrap = shallow(<Map latitude={dummyLat} longitude={dummyLon} whereData={dummyWhereData} />);
  
  it('should be defined', () => {
    expect(Map).toBeDefined();
  });
 
  it('should render a Map component', () => {
    expect(mapWrap).toMatchSnapshot();
  });
  
  it('should insert a hyperlink to Google Maps', () => {
    const href = mapWrap.find('a').props().href;
    expect(href).toBe('https://www.google.com/maps/search/?api=1&query=30%20Nowhere%20St.%2C%20Sometown%2C%20CA');
  });
  
  it('should insert a map img', () => {
    const src = mapWrap.find('img').props().src;
    const expectedSrc = `https://maps.googleapis.com/maps/api/staticmap?key=${googleMapsKey}&center=37.75,-122.419998&size=480x300&zoom=17&scale=2&markers=color%3Ared%7Csize%3Alarge%7C`;
    expect(src).toBe(expectedSrc);
  });
});
