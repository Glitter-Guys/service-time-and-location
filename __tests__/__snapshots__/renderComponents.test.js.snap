// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Map /> should render a Map component 1`] = `
<div
  className="map"
>
  <a
    className="map__link"
    href="https://www.google.com/maps/search/?api=1&query=30%20Nowhere%20St.%2C%20Sometown%2C%20CA"
    target="_blank"
  >
    <img
      alt="map of location"
      className="map__img"
      src="https://maps.googleapis.com/maps/api/staticmap?key=AIzaSyC54lAZfmFGtXLamlbYzwabbOMHFEdqGwM&center=37.75,-122.419998&size=480x300&zoom=17&scale=2&markers=color%3Ared%7Csize%3Alarge%7C"
    />
  </a>
</div>
`;

exports[`<When /> should render a When component 1`] = `
<div
  className="when"
>
  <div
    className="when__icon"
  >
    <InlineSVG
      cacheGetRequests={false}
      height="24"
      onLoad={[Function]}
      src="./icons/clock.svg"
      supportTest={[Function]}
      uniquifyIDs={true}
      viewBox="0 0 24 24"
      width="24"
      wrapper={[Function]}
    />
  </div>
  <div
    className="when__text"
  >
    <span>
      <span>
        <t
          ago={false}
          calendar={false}
          decimal={false}
          element="time"
          filter={[Function]}
          format="dddd, MMMM D YYYY"
          fromNow={false}
          interval={60000}
          onChange={[Function]}
          toNow={false}
          unit={null}
          unix={false}
          utc={false}
        >
          2018-03-13T02:00:00Z
        </t>
      </span>
      <t
        ago={false}
        calendar={false}
        decimal={false}
        element="time"
        filter={[Function]}
        format="h:mm A"
        fromNow={false}
        interval={60000}
        onChange={[Function]}
        toNow={false}
        unit={null}
        unix={false}
        utc={false}
      >
        2018-03-13T02:00:00Z
      </t>
       to 
      <t
        ago={false}
        calendar={false}
        decimal={false}
        element="time"
        filter={[Function]}
        format="h:mm A"
        fromNow={false}
        interval={60000}
        onChange={[Function]}
        toNow={false}
        unit={null}
        unix={false}
        utc={false}
      >
        2018-03-13T04:00:00Z
      </t>
    </span>
    <span
      className="when__series"
    >
      Repeats every week
    </span>
  </div>
</div>
`;
